<launch>

    <!-- SENSORS -->
    <node name="left_encoder" pkg="crues_sensors" type="enc_node.py">
        <remap from="~rate" to="encoder/rate" />
        <remap from="~pin_a" to="pins/ela" />
        <remap from="~pin_b" to="pins/elb" />
        <remap from="/wheel" to="/lwheel"/>
    </node>

    <!-- ACTUATORS -->
    <node name="motor_sleep" pkg="crues_actuators" type="motor_sleep_node.py" />
    <node name="left_motor" pkg="crues_actuators" type="drive_node.py">
        <param name="range_min" type="double" value="-0.5" />
        <param name="range_max" type="double" value="0.5" />
        <remap from="~pwm_pin" to="pins/mlp" />
        <remap from="~dir_pin" to="pins/mld" />
        <remap from="/motor_cmd" to="/lmotor_cmd" />
    </node>

    <!-- CONTROL -->
    <node name="lpid_velocity" pkg="differential_drive" type="pid_velocity.py">
        <param name="out_min" type="double" value="-0.5" />
        <param name="out_max" type="double" value="0.5" />
        <param name="rolling_pts" type="int" value="2" />
        <remap from="~Kp" to="pid/Kp" />
        <remap from="~Ki" to="pid/Ki" />
        <remap from="~Kd" to="pid/Kd" />
        <remap from="~rate" to="encoder/rate" />
        <remap from="/wheel" to="/lwheel" />
        <remap from="/motor_cmd" to="/lmotor_cmd" />
        <remap from="/wheel_vtarget" to="/lwheel_vtarget" />
        <remap from="/wheel_vel" to="/lwheel_vel" />
    </node>

</launch>